AStarAlgorithmsProject/Alg Scripts/AStarSolver.cs

```diff
@@ -0,0 +1,231 @@
+            { get { return Point.Distance(location, Tile.Goal.location); } }
+            public double Distance2Start // the "g(x)" value of the huristics. Or "cost to get here"
+            { get { return Point.Distance(location, Tile.Start.location); } }
+            public double Cost // the "f(x)" value of the heuristic. Or "cost of path"
+            { get { return Distance2Start*costScalar + Distance2Goal; } }
+
+        }
+
+        private class SortTileByF : IComparer<Tile>
+        {
+            public int Compare(Tile t1, Tile t2) // if you want to have some fun, swap these values so that A* will find the most direct but least optimal solution.
+            {
+                if (t1.Cost > t2.Cost)
+                {
+                    return 1;
+                }
+                else if (t1.Cost < t2.Cost)
+                {
+                    return -1;
+                }
+                else
+                {
+                    return 0;
+                }
+            }
+        }  // Simple class to be used by the Sort method of the List class. 
+
+        public bool ValidTile(int x, int y)
+        {
+            return (x >= 0 && x < AStarSolver.MapSize) && (y >= 0 && y < AStarSolver.MapSize);
+        } // returns true if the tile at the given location is within the bounds of the map
+
+        public bool ValidTile(Point p)
+        {
+            return (p.X >= 0 && p.X < AStarSolver.MapSize) && (p.Y >= 0 && p.Y < AStarSolver.MapSize);
+        } // overload of ValidTile to use a passed point parameter
+
+        //Builds the traversal path bassed on the most ideal canidate of the current tile
+        private void TracePath() 
+        {
+            List<Tile> result = new List<Tile>();
+            int nx = 0;
+            int ny = 0;
+
+            foreach (Point p in Directions) // looks at a possible cordinate for every tile neighboring the current tile
+            {
+                nx = Tile.Current.Location.X + p.X;
+                ny = Tile.Current.Location.Y + p.Y;
+
+                if (ValidTile(nx, ny)) // continue to evaluate only if the tile is on the map
+                {
+                    Tile neighbor = map[nx, ny]; // the tile being evaluated
+                    if (neighbor.Passable) // continue to evaluate only if it's a tile that can be mvoed to / on
+                    {
+                        if (neighbor.State != TileStates.Closed) // continue to evaluate if the tile has not already been closed 
+                        {
+                            if (neighbor.State == TileStates.Unchecked) // if it hasnt been checked before, we need only add it to the list of open tiles 
+                            {
+                                neighbor.State = TileStates.Open;
+                                neighbor.Parent = map[Tile.Current.Location.X, Tile.Current.Location.Y]; // go set its parent to be the first tile to disover it by default
```

Comment:
lastNode or some other name might make for a better name. I was just thrown off by 'Parent' but I don't have a great alternative off of the top of my head.

Reply:
But it's not quite a "lastNode" of a tile, it's the prior tile of a given tile in the tree we are making. If you feel it would be better, go ahead and change it yourself, but I feel the current name is apt.